model/src/main/proto/survey.proto

```diff
@@ -0,0 +1,182 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "org.oppia.android.app.model";
+option java_multiple_files = true;
+
+// Corresponds to a single survey.
+message Survey{
+  // The ID of the survey.
+  string survey_id = 1;
+
+  // A list of questions that make up the survey.
+  repeated SurveyQuestion questions = 2;
+}
+
+// Represents a question that is part of the Survey.
+message SurveyQuestion{
+  // The ID of the survey question.
+  string question_id = 1;
+
+  // The type of question.
+  QuestionType question_type = 2;
+
+  // The actual text value of the question.
+  string  question = 3;
+
+  // The semantic name of the question used to identify it in the analytics dashboard.
+  QuestionName question_name = 4;
+
+  // A list of multi-choice answer options for a given question.
+  repeated  SurveyQuestionAnswer answer_list = 5;
```

Comment:
It is my understanding that the `answer_list` field can be left empty, as the respondent will provide their answer as free-form text.

It is not clear to me why a oneof _answer_ would be needed, since I have modelled the question around QuestionType.  Could you be suggesting that instead of modelling around QuestionType, I might consider using oneof _answer_ ?

Reply:
One possible issue is that an empty proto value (whether it be text or answer_list) is ambiguous--does it mean "this isn't a multiple choice" or "there aren't any choices defined for this multiple choice question"? It makes it harder to reason about a model if it can take different forms but share properties that should be ignored for certain types (this can result in extra unnecessary error checking needing to be added, or potential for conceptual mix-ups later on).

In this case, you could replace QuestionType with a ``oneof`` and better separate the properties relevant to each type of question, e.g.:

```proto
message SurveyQuestion {
  string question_id = 1;
  QuestionName question_name = 2;
  oneof question_type {
    SurveyQuestionOptionList multiple_choice = 3;
    SurveyQuestionOptionList multiple_choice_with_other = 4;
    bool free_form_text = 5;
  }
}

message SurveyQuestionOptionList {
  repeated SurveyQuestionOption options = 1;
}
```

(Note the extra ``SurveyQuestionOptionList`` is needed because proto doesn't currently support having repeated values in ``oneof``s).

Note how in the case of free-form text we now no longer have to even question what to do with the multiple choice options as a field because the field can't exist in that case.

Reply:
Your proposed `SurveyQuestion` message structure using `oneof` looks good. In this case, I can create a seperate UiModel for QuestionType which will be used for rendering.